app/services/idv/analytics_events_enhancer.rb

```diff
@@ -134,9 +134,31 @@ def self.should_enhance_method?(method_name)
     def common_analytics_attributes
       {
         proofing_components: proofing_components,
+        active_profile_idv_level: active_profile&.idv_level,
+        pending_profile_idv_level: pending_profile&.idv_level,
+        profile_history: profile_history,
       }.compact
     end
 
+    def active_profile
+      return if !user&.respond_to?(:active_profile) || !user.active_profile
+      user.active_profile
+    end
+
+    def pending_profile
+      return if !user&.respond_to?(:pending_profile) || !user.pending_profile
+      user.pending_profile
+    end
+
+    def profile_history
+      return if !user&.respond_to?(:profiles)
+
+      (user&.profiles || []).
+        sort_by { |profile| profile.created_at }.
+        map { |profile| ProfileLogging.new(profile) }.
+        presence
```

Comment:
do we know the kinds of queries we're going to make against this profiles array? are we going to be hardcoding indexes?

Reply:
I don't think the use case will be querying against this array--the data is just there for forensic purposes.

Reply:
I feel like in that case... it would be a good reason to maybe consider not logging? we'll have it in the DB?

Reply:
Or maybe log one snapshot like early on in the process and one snapshot at the end of the process?

Reply:
I think that is fair, yeah. I think there is a desire to to have `active_profile_idv_level` and `pending_profile_idv_level` available throughout as a way of monitoring biometric rollout. But we could probably just add `profile_history` to a subset of events, maybe:

* `IdV: intro visited` (i.e. the start of idv)
* `IdV: start over`
* `IdV: doc auth verify proofing results`
* `IdV: final resolution`
* `IdV: Verify please call visited`